specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-02-01/managedClusters.json

```diff
@@ -1952,6 +1952,13 @@
           "$ref": "#/definitions/ManagedClusterAADProfile",
           "description": "Profile of Azure Active Directory configuration."
         },
+        "autoScalerProfile": {
+          "type": "object",
+          "additionalProperties": {
```

Comment:
>additionalProperties [](start = 11, length = 20)

This lets the user specify arbitrary properties. How does the user know what they are allowed to set? Is it possible to create a schema for it?

Reply:
The validation will happen server-side. The allowed parameters will evolve over-time.

Reply:
The typical pattern is to strongly model all properties that the user has to specify. There are many platform capabilities that will suffer when you rely on `additionalProperties`. Here are some examples off the top of my head:
- Policy would never be able to enforce autoscaler settings. 
- ARM template deployment what-if won't be able to understand patterns in the properties and will produce noisy diffs.
- The JSON schema that will get generated for your resources will make authoring of ARM templates in VS code more difficult because IntelliSense won't provide suggestions for autoscaler settings.

Why are we deviating from that model? We have mechanisms to allow APIs to evolve, so change over time is definitely possible.

Reply:
Discussed with @marwanad over internal IM. The settings are passed as-is to a Kubernetes add-on, which limits strongly modeling this part of the API because it is versioned independently from ARM API version.

Reply:
We've decided to explicitly have the supported parameters defined in the spec.